static/js/actions/collectionUi.js

```diff
@@ -26,3 +33,14 @@ export const setAdminLists = createAction(SET_ADMIN_LISTS);
 
 export const SET_SELECTED_VIDEO_KEY = qualifiedName('SET_SELECTED_VIDEO_KEY');
 export const setSelectedVideoKey = createAction(SET_SELECTED_VIDEO_KEY);
+
+export const SET_IS_NEW = qualifiedName('SET_IS_NEW');
+export const setIsNew = createAction(SET_IS_NEW);
+
+export const startNewCollectionDialog = (collection: ?Collection = null) =>
```

Comment:
One nice feature we had before this PR was that the forms state was maintained when you opened/closed the form without submitting it or selecting a different collection. That wasn't specifically asked for, and it's kind of an edge case that someone would need it with this relatively small form, but it's worth noting we do lose that with this approach. Either way, I would argue that launching the new collection dialog doesn't need to do the form init at all. We can just clear it after submission. Maybe it's better to create 2 different actions here: `showNewCollectionDialog` and `showEditCollectionDialog`. `showNewCollectionDialog` would avoid that `initCollectionForm` call entirely

Reply:
Sure